Fix panic in PAC controller on delete tag event in GitLab

zakisk opened the pull request:
fixed panic in PAC controller when a tag is deleted on gitlab.

https://issues.redhat.com/browse/SRVKP-6636

https://redhat-internal.slack.com/archives/C04PZ7H0VA8/p1727708590297089

# Changes <!-- 🎉🎉🎉 Thank you for the PR!!! 🎉🎉🎉 -->

<!-- Describe your changes here- ideally you can get that description straight from
your descriptive commit message(s)! -->

# Submitter Checklist

- [ ] 📝 Please ensure your commit message is clear and informative. For guidance on crafting effective commit messages, refer to the How to write a git commit message guide. We prefer the commit message to be included in the PR body itself rather than a link to an external website (ie: Jira ticket).

- [ ] ♽ Before submitting a PR, run make test lint to avoid unnecessary CI processing. For an even more efficient workflow, consider installing [pre-commit](https://pre-commit.com/) and running pre-commit install in the root of this repository.

- [ ] ✨ We use linters to maintain clean and consistent code. Please ensure you've run make lint before submitting a PR. Some linters offer a --fix mode, which can be executed with the command make fix-linters (ensure [markdownlint](https://github.com/DavidAnson/markdownlint) and [golangci-lint](https://github.com/golangci/golangci-lint) tools are installed first).

- [ ] 📖 If you're introducing a user-facing feature or changing existing behavior, please ensure it's properly documented.

- [ ] 🧪 While 100% coverage isn't a requirement, we encourage unit tests for any code changes where possible.

- [ ] 🎁 If feasible, please check if an end-to-end test can be added. See [README](https://github.com/openshift-pipelines/pipelines-as-code/blob/main/test/README.md) for more details.

- [ ] 🔎 If there's any flakiness in the CI tests, don't necessarily ignore it. It's better to address the issue before merging, or provide a valid reason to bypass it if fixing isn't possible (e.g., token rate limitations).

chmouel commented:
seems like your test is failing on --- FAIL: TestGitlabDeleteTagEvent (16.82s)
 have you tested it ? 

```
    gitlab_delete_tag_event_test.go:49: assertion failed: error is not nil: could not find a match using the labelSelector: app.kubernetes.io/name=controller in container pac-controller for regexp: event Delete Tag Push Hook is not supported*
```

zakisk commented:
@chmouel yes, tested on my local machine and it was working fine.

zakisk committed:
Fix panic in PAC controller on delete tag event in gitlab

fixed panic in PAC controller when a tag delete event
occurs in a repository on gitlab. E2E test is added
as well to confirm this behaviour.

https://issues.redhat.com/browse/SRVKP-6636

Signed-off-by: Zaki Shaikh <zashaikh@redhat.com>

zakisk commented:
/test

chmouel commented on test/gitlab_delete_tag_event_test.go:
```diff
@@ -0,0 +1,55 @@
+	assert.NilError(t, err)
+	runcnx.Clients.Log.Infof("Created Tag %s in %s repository", tagName, projectinfo.Name)
+
+	err = tgitlab.DeleteTag(glprovider.Client, projectinfo.ID, tagName)
```
if there is going to be a issue with the tag creation the tag will be kept on the repo? is there an issue if the tag is already on the repo wand we try to recreate it on the next test?

zakisk replied:
@chmouel changed the way tag is created.

chmouel replied:
can you add a `defer` to make sure the tag is deleted at the end of the run ?
